nima/webclient/webclient/src/main/java/io/helidon/nima/webclient/http/spi/SourceHandler.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.nima.webclient.http.spi;
+
+import io.helidon.common.GenericType;
+import io.helidon.nima.webclient.ClientResponse;
+
+/**
+ * {@link java.util.ServiceLoader} provider interface for {@link Source} handlers.
+ *
+ * @param <T> event type
+ * @param <X> type of source
+ */
+public interface SourceHandler<T, X extends Source<T>> {
```

Comment:
I prefer to use names that end with `Provider` when this is an SPI through `ServiceLoader`, so maybe `SourceHandlerProvider` or even `EventSourceHandlerProvider`